java/ingest/ingest-runner/src/main/java/sleeper/ingest/impl/recordbatch/arrow/ArrowRecordBatchFactory.java

```diff
@@ -196,6 +183,21 @@ public ArrowRecordBatchFactory<Record> buildAcceptingRecords() {
         }
 
         public ArrowRecordBatchFactory<T> build() {
+            Objects.requireNonNull(schema, "schema must not be null");
+            Objects.requireNonNull(localWorkingDirectory, "localWorkingDirectory must not be null");
+            Objects.requireNonNull(recordWriter, "recordWriter must not be null");
```

Comment:
It's hard to see which fields should be validated here. Can this be moved to the top of the builder class just below the constructor, next to the fields?